File: Esercizi C/Algebra/Cerchio.c
/*
Dati in input: - Raggio di un cerchio;
Calcolare la circonferenza, l'area, il diametro e visualizzare i risultati. Pigreco = M_PI
*/

#include <stdio.h>

#include <math.h> //libreria delle funzioni matematiche


int main(){
	double r;
	
	printf("Raggio: ");
	scanf("%lf", &r);
	
	double c = 2*M_PI*r;
	printf("\nCirconferenza: %.2lf.", c);
	double a = r*r*M_PI;
	printf("\nArea: %.2lf", a);
	double d = r*2;
	printf("\nDiametro: %.2lf", d);
	
}

File: Esercizi C#/Classi/ClasseCerchio/Cerchio.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClasseCerchio
{
    class Cerchio
    {
        private readonly double raggio;

        public Cerchio(double raggio)
        {
            this.raggio = raggio;
        }

        public double Diametro()
        {
            return raggio * 2;
        }

        public double Area()
        {
            return Math.PI * raggio * raggio;
        }

        public double Circonferenza()
        {
            return 2 * Math.PI * raggio;
        }

        public string StampaDettaglio()
        {
            string msg = $"Dati del cerchio: " +
                $"\nRaggio: {raggio}" +
                $"\nDiametro: {Diametro()}" +
                $"\nArea: {Area()}" +
                $"\nCirconferenza: {Circonferenza()}";
            return msg;
        }
        public string StampaLineare()
        {
            string msg = $"Dati del cerchio: " +
                $" Raggio: {raggio}," +
                $" Diametro: {Diametro()}," +
                $" Area: {Area()}," +
                $" Circonferenza: {Circonferenza()}";
            return msg;
        }

    }
}
